src/develop/imageop_gui.c

```diff
@@ -0,0 +1,219 @@
+  {
+    if (self->gui_changed) self->gui_changed(self, combobox, &previous);
+
+    dt_iop_color_picker_reset(self, TRUE);
+
+    dt_dev_add_history_item(darktable.develop, self, TRUE);
+  }
+}
+
+GtkWidget *dt_bauhaus_slider_new_from_params_box(dt_iop_module_t *self, const char *param)
+{
+  dt_iop_params_t *p = (dt_iop_params_t *)self->params;
+  dt_introspection_field_t *f = self->so->get_f(param);
+
+  GtkWidget *slider = NULL;
+  gchar *str;
+
+  if (f && f->header.type == DT_INTROSPECTION_TYPE_FLOAT)
+  {
+    float min = f->Float.Min;
+    float max = f->Float.Max;
+    float defval = *(float*)self->so->get_p(p, param);
+    int digits = 2;
+    float step;
+
+    float top = fmaxf(fabsf(min),fabsf(max));
+    if (top>=100)
+    {
+      step = 1.f;
+    }
+    else
+    {
+      step = top / 100;
+      float log10step = log10f(step);
+      float fdigits = floorf(log10step+.1);
+      step = powf(10.f,fdigits);
+      if (log10step - fdigits > .5)
+        step *= 5;
+      if (fdigits < -2.f) 
+        digits = -fdigits;
+    }
+
+    slider = dt_bauhaus_slider_new_with_range_and_feedback(self, min, max, step, defval, digits, 1);
+
+    if (*f->header.description)
+    {
+      dt_bauhaus_widget_set_label(slider, NULL, gettext(f->header.description));
+    }
+    else
+    {
+      str = dt_util_str_replace(f->header.field_name, "_", " ");
+    
+      dt_bauhaus_widget_set_label(slider, NULL, gettext(str));
+
+      g_free(str);
+    }
+
+    const char *post = ""; // set " %%", " EV" etc
+
+    if (min < 0 || (post && *post))
+    {
+      str = g_strdup_printf("%%%s.0%df%s", (min < 0 ? "+" : ""), digits, post);
+
+      dt_bauhaus_slider_set_format(slider, str);
+    
+      g_free(str);
+    }
+
+    dt_module_param_t *module_param = (dt_module_param_t *)g_malloc(sizeof(dt_module_param_t));
+    module_param->module = self;
+    module_param->param_offset = self->so->get_p(p, param) - p;
+    g_signal_connect_data(G_OBJECT(slider), "value-changed", G_CALLBACK(generic_slider_callback), module_param, (GClosureNotify)g_free, 0);
+  }
+  else
+  {
+    str = g_strdup_printf("'%s' is not a float/slider parameter", param);
+
+    slider = GTK_WIDGET(GTK_LABEL(gtk_label_new(str)));
+
+    g_free(str);
+  }
+
+  gtk_box_pack_start(GTK_BOX(self->widget), slider, FALSE, FALSE, 0);
+
+  return slider;
+}
+
+GtkWidget *dt_bauhaus_combobox_new_from_params_box(dt_iop_module_t *self, const char *param)
+{
+  dt_iop_params_t *p = (dt_iop_params_t *)self->params;
+  dt_introspection_field_t *f = self->so->get_f(param);
+
+  GtkWidget *combobox = NULL;
+  gchar *str;
+
+  if (f && (f->header.type == DT_INTROSPECTION_TYPE_ENUM || f->header.type == DT_INTROSPECTION_TYPE_INT))
+  {
+    combobox = dt_bauhaus_combobox_new(self);
+
+    if (*f->header.description)
+    {
+      dt_bauhaus_widget_set_label(combobox, NULL, gettext(f->header.description));
+    }
+    else
+    {
+      str = dt_util_str_replace(f->header.field_name, "_", " ");
```

Comment:
The fact that the introspection filed is transformed to form the name of the combo by removing the _ should be documented somewhere.

Reply:
My plan is to significantly update "useless.c" once I've submitted (and received feedback on) a number of the non-trivial modules. That might be a good place to document this sort of thing would you agree? Happy to improve documentation elsewhere as well.

Also, as I now commented in the code, I'd prefer this transformation to happen in the introspection code, so it would also be available to all other users of that data. I've mentioned this to @houz on IRC but I'm not sure if he agreed :-) It would save quite a lot of trivial $DESCRIPTION comments.